src/main/io/rcdevice.h

```diff
@@ -0,0 +1,227 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "common/streambuf.h"
+#include "drivers/serial.h"
+
+//
+// The protocol for Runcam Device definition
+//
+#define RCDEVICE_PROTOCOL_HEADER                            0xCC
+
+#define RCDEVICE_PROTOCOL_MAX_PACKET_SIZE                   64
+#define RCDEVICE_PROTOCOL_MAX_DATA_SIZE                     62
+#define RCDEVICE_PROTOCOL_MAX_MENUITEM_PER_PAGE             32
+#define RCDEVICE_PROTOCOL_MAX_SETTING_NAME_LENGTH           20
+#define RCDEVICE_PROTOCOL_MAX_SETTING_VALUE_LENGTH          20
+#define RCDEVICE_PROTOCOL_MAX_CHUNK_PER_RESPONSE            5
+#define RCDEVICE_PROTOCOL_MAX_TEXT_SELECTIONS               12
+
+// Commands
+#define RCDEVICE_PROTOCOL_COMMAND_GET_DEVICE_INFO           0x00
+// camera control
+#define RCDEVICE_PROTOCOL_COMMAND_CAMERA_CONTROL            0x01
+// 5 key osd cable simulation
+#define RCDEVICE_PROTOCOL_COMMAND_5KEY_SIMULATION_PRESS     0x02
+#define RCDEVICE_PROTOCOL_COMMAND_5KEY_SIMULATION_RELEASE   0x03
+#define RCDEVICE_PROTOCOL_COMMAND_5KEY_CONNECTION           0x04
+// device setting access
+#define RCDEVICE_PROTOCOL_COMMAND_GET_SETTINGS              0x10
+#define RCDEVICE_PROTOCOL_COMMAND_READ_SETTING_DETAIL       0x11
+#define RCDEVICE_PROTOCOL_COMMAND_READ_SETTING              0x12
+#define RCDEVICE_PROTOCOL_COMMAND_WRITE_SETTING             0x13
+// display port support
+#define RCDEVICE_PROTOCOL_COMMAND_DISP_FILL_REGION          0x20
+#define RCDEVICE_PROTOCOL_COMMAND_DISP_WRITE_CHAR           0x21
+#define RCDEVICE_PROTOCOL_COMMAND_DISP_WRITE_HORT_STRING    0x22
+#define RCDEVICE_PROTOCOL_COMMAND_DISP_WRITE_VERT_STRING    0x23
```

Comment:
Use `RCDEVICE_PROTOCOL_COMMAND_DISP_WRITE_HORIZONTAL_STRING` and `RCDEVICE_PROTOCOL_COMMAND_DISP_WRITE_VERTICAL_STRING`

Reply:
yes, i get it